src/Surface.zig

```diff
@@ -1540,6 +1542,23 @@ fn modsChanged(self: *Surface, mods: input.Mods) void {
     }
 }
 
+/// Hide the mouse if `mouse-hide-after` is configured and the mouse has been
+/// idle (no movement) for at least that duration.
+pub fn maybeHideMouseAfterIdle(self: *Surface) void {
```

Comment:
Do we need this core stuff at all? I think it might be cleaner to just implement this 100% in the apprt.